via/services/google_drive.py

```diff
@@ -16,29 +17,48 @@
 LOG = getLogger(__name__)
 
 
-def translate_google_error(
-    error,
-):  # pylint:disable=too-many-return-statements
+class GoogleDriveErrorSchema(Schema):
+    """Schema for the JSON bodies of Google Drive error responses."""
+
+    class Meta:
+        unknown = INCLUDE
+
+    class Errors(Schema):
+        class Error(Schema):
+            message = fields.String()
+            reason = fields.String()
+
+        errors = fields.List(
+            fields.Nested(Error(unknown=INCLUDE)),
+            required=True,
+            validate=validate.Length(min=1),
+        )
+
+    @pre_load
+    def pre_load(self, data, **_kwargs):  # pylint:disable=no-self-use
+        try:
+            return data.response.json()
+        except JSONDecodeError as err:
+            raise ValidationError("Invalid JSON") from err
+
+    error = fields.Nested(Errors(unknown=INCLUDE), required=True)
```

Comment:
We talked about this on slack, but I thought it might be good to capture it on the ticket.

I think it's simpler and better to keep the execution flow into the schemas as a sort of dead end, rather than through road for the data. We use Marshmallow `load()` to "transform" the data, but in truth the output is almost always the same as the input. So our transmformation is really just the identity transformation. I think we might be better off calling `validate()` instead as it advertises to the reader that the schema is side-effectless and can be ignored as long as it passes.

So:

```
data = MySchema().load(request)
... # use data
```

To:

```
data = request.json
MySchema().validate(data)
... # use data
```

Reply:
This is covered in https://github.com/hypothesis/via/pull/643#discussion_r726058550 below. I tried it but I ended up with more lines of code and triggering PyLint warnings to boot :man_shrugging: You'd need to do more work on `translate_google_error()` to enable it. That function is too long and complex. Maybe something for a separate PR